### ISSUE TITLE
Imported types cause false positives with `no-unused-vars` rule

### ISSUE THREAD

**Issue by Hilzu** [eslint/typescript-eslint-parser#457]
**What version of TypeScript are you using?**
2.7.2

**What version of `typescript-eslint-parser` are you using?**
14.0.0

**What code were you trying to parse?**
```ts
import { Server } from "http"
import app from "./app"

let server: Server | null = null

server = app.listen(() => {
  console.log("Listening on", server!.address())
})
```

**What did you expect to happen?**
No errors are reported with the `no-unused-vars` rule on

**What happened?**
```
src/test.ts
  1:10  error    'Server' is defined but never used  no-unused-vars
```

The `no-unused-vars` rule was previously mentioned in the readme as having known issues. Now that it's removed I expected the above code to work.

**Comment by kinyat** [comment]
We cannot turn on `noUnusedLocals` on local development as it is very annoying to have this blocking the complier.

Would really love to rely on eslint to check to allow compiling.

**Comment by tushar-singh** [comment]
I have this issue as well. Using the same workaround as kinyat via overrides (so I can still have these rules enabled in `.js` and `.jsx` files)
```
  "rules": { ... },
  "overrides": [
    {
      "files": ["**/*.ts", "**/*.tsx"],
      "rules": {
        "no-unused-vars": ["off"],
        "no-undef": ["off"]
      }
    }
  ]
```

**Comment by JamesHenry** [comment]
There is already a custom rule in the [plugin](https://github.com/nzakas/eslint-plugin-typescript) for `no-unused-vars`: https://github.com/nzakas/eslint-plugin-typescript/blob/master/docs/rules/no-unused-vars.md

If that rule is no sufficient for this use-case then a bug report should be filed against it, and ideally a PR submitted to extend at.

Thanks!

**Comment by KaelWD** [comment]
I'm using that, it fails if you import types that don't have a value. 

```js
'no-unused-vars': 'error',
'typescript/no-unused-vars': 'error',
```

```ts
// ERROR: 'Options' is defined but never used
import { VuetifyUseOptions as Options } from 'types'
...
// Even though I use it here as a type
export default function lang (config: Options['lang'] = {}): VuetifyLanguage {
```

**Comment by JamesHenry** [comment]
As per my previous message, a bug report should be filed against the plugin, this is not a parsing issue

**Comment by KaelWD** [comment]
Forgot this was a separate repo 😅
